core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -615,7 +616,11 @@ public void register(String name, Suggester<?> t) {
         highlighters.register("fvh",  new FastVectorHighlighter(settings));
         highlighters.register("plain", new PlainHighlighter());
         highlighters.register("postings", new PostingsHighlighter());
-
+        UnifiedHighlighter uh = new UnifiedHighlighter();
+        highlighters.register("unified", uh);
+        highlighters.register("unified_plain", uh);
+        highlighters.register("unified_postings", uh);
+        highlighters.register("unified_fvh", uh);
```

Comment:
oh I see then we have a switch on the highlighter type in the impl, maybe write a comment about it here?